crates/ruff_python_ast/src/identifier.rs

```diff
@@ -251,10 +228,13 @@ impl<'a> IdentifierTokenizer<'a> {
         while let Some(c) = self.cursor.bump() {
             match c {
                 c if is_python_identifier_start(c) => {
-                    let start = self.offset.add(self.cursor.offset()).sub(c.text_len());
+                    self.offset += self.cursor.token_len();
+                    self.cursor.start_token();
```

Comment:
We also need to test that it is not a string prefix.

Reply:
This tokenizer makes a lot of assumptions, one of which is that you're parsing code that can't contain strings (e.g., the `else` in `else:`).